scripts/asm-installer/install_asm

```diff
@@ -707,27 +713,108 @@ check_istio_not_deployed(){
 
 validate_istio_version() {
   info "Checking existing Istio version(s)..."
-  local OUTPUT
-  OUTPUT="$(retry 3 ./istio-${RELEASE}/bin/istioctl version -o json)"
-  local FOUND_VALID_VERSION
-  FOUND_VALID_VERSION="false"
-  for version in $(echo "${OUTPUT}" | jq -r '.meshVersion[].Info.version' -r); do
+  local VERSION_OUTPUT; VERSION_OUTPUT="$(retry 3 ./${ISTIOCTL_REL_PATH} version -o json)"
+  local FOUND_VALID_VERSION; FOUND_VALID_VERSION=0
+  for version in $(echo "${VERSION_OUTPUT}" | jq -r '.meshVersion[].Info.version' -r); do
     if [[ "$version" =~ ^$RELEASE_LINE ]]; then
       info "  $version (suitable for migration)"
-      FOUND_VALID_VERSION="true"
-    fi
-    if ! [[ "$version" =~ ^$RELEASE_LINE ]]; then
+      FOUND_VALID_VERSION=1
+    else
       info "  $version (not suitable for migration)"
     fi
     if [[ "$version" =~ "asm" ]]; then
       fatal "Cannot migrate from version $version. Only migration from OSS Istio to the ASM distribution is supported."
     fi
   done
-  if [[ "$FOUND_VALID_VERSION" != "true" ]]; then
+  if [[ "$FOUND_VALID_VERSION" -eq 0 ]]; then
     fatal "Migration requires an existing control plane in the ${RELEASE_LINE} line."
   fi
 }
 
+validate_asm_version() {
+  info "Checking existing ASM version(s)..."
+  local VERSION_OUTPUT; VERSION_OUTPUT="$(retry 3 ./${ISTIOCTL_REL_PATH} version -o json)"
+  local FOUND_INVALID_VERSION; FOUND_INVALID_VERSION=0
+  for VERSION in $(echo "${VERSION_OUTPUT}" | jq -r '.meshVersion[].Info.version'); do
+    if ! [[ "${VERSION}" =~ "asm" ]]; then
+      fatal "Cannot upgrade from version ${VERSION}. Only upgrade from ASM distribution is supported."
+    fi
+
+    if version_valid_for_upgrade "${VERSION}"; then
+      info "  ${VERSION} (suitable for migration)"
+    else
+      info "  ${VERSION} (not suitable for migration)"
+      FOUND_INVALID_VERSION=1
+    fi
+  done
+  if [[ "$FOUND_INVALID_VERSION" -eq 1 ]]; then
+    fatal "Upgrade requires all existing control planes to be between versions 1.6.0-asm.0 (inclusive) and ${RELEASE} (exclusive)."
```

Comment:
Can you change 1.6.0 to 1.$((MINOR-1)).0?

Reply:
sg